src/main/scala/com/databricks/spark/redshift/FilterPushdown.scala

```diff
@@ -46,6 +46,7 @@ private[redshift] object FilterPushdown {
      getTypeForAttribute(schema, attr).map { dataType =>
        val sqlEscapedValue: String = dataType match {
          case StringType => s"\\'${value.toString.replace("'", "\\'\\'")}\\'"
+         case DateType | TimestampType => s"\\'$value\\'"
```

Comment:
I think this might actually be subtly wrong for two reasons:

1. the internal representation of these types might not be Java date or timestamp types; rather, it might be some type of numeric primitive, in which case this would return the wrong result.
2. Java dates and timestamps might have locale-sensitive toString() methods; we should explicitly use a formatter when converting them to strings.